File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/playerstate/PersonalWorldState.java
package top.likoslupus.cellulosesz.api.playerstate;

import static java.util.Objects.requireNonNull;

/**
 * Typed personal world overrides, independent of the legacy persisted representation.
 */
public record PersonalWorldState(
        PersonalTimeSetting time,
        PersonalWeatherSetting weather
) {

    public PersonalWorldState {
        requireNonNull(time, "time");
        requireNonNull(weather, "weather");
    }

    public static PersonalWorldState reset() {
        return new PersonalWorldState(PersonalTimeSetting.reset(), PersonalWeatherSetting.RESET);
    }

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/playerstate/PersonalWorldState.kt
package top.likoslupus.cellulosesz.api.playerstate

/**
 * Typed personal world overrides, independent of the legacy persisted representation.
 */
@JvmRecord
public data class PersonalWorldState(
    public val time: PersonalTimeSetting,
    public val weather: PersonalWeatherSetting
) {

    public companion object {

        @JvmStatic
        public fun reset(): PersonalWorldState =
            PersonalWorldState(
                PersonalTimeSetting.reset(),
                PersonalWeatherSetting.RESET
            )

    }

}
